parsl/config.py

```diff
@@ -125,11 +128,9 @@ def __init__(self,
     def executors(self) -> Sequence[ParslExecutor]:
         return self._executors
 
-    @executors.setter
-    def executors(self, executors: Sequence[ParslExecutor]):
+    def _validate_executors(self, executors: Sequence[ParslExecutor]) -> None:
         labels = [e.label for e in executors]
         duplicates = [e for n, e in enumerate(labels) if e in labels[:n]]
         if len(duplicates) > 0:
             raise ConfigurationError('Executors must have unique labels ({})'.format(
                 ', '.join(['label={}'.format(repr(d)) for d in duplicates])))
```

Comment:
From a transparency in the non-happy-path case, perhaps consider checking the `len(executors) == 0` case as well?

Reply:
see db9d890 although it raises a weird use case i hadn't thought of before...